### PR TITLE
Composer cleanup

### PR DESCRIPTION
**Pull request by ghost**
```
 Moves psr/cache to a placeholder repository so it can be easily replaced in the future.

Adds phpunit, php_codesniffer, phpmd, and additional php_unit modules to the require-dev section
```

Precursor for moving unit test and code sniff execution for an external directory to using the composer installed libraries.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -4,12 +4,27 @@
     "keywords": ["joomla","framework"],
     "homepage": "https://github.com/joomla/joomla-framework",
     "license": "GPL-2.0+",
+	"repositories": [
+		{
+			"type": "vcs",
+			"url": "https://github.com/garyamort/PsrCache.git"
```
Comment (eddieajau) [inline composer.json:10]: I'm not sure I like this going to an adhoc repository. However, we should definitely move it to our own specification until the PSR is ratified officially (but that will technically cause some b/c headaches in the mean time). I think this is probably a V2 fix?

Comment (dongilbert) [inline composer.json:10]: I would suggest moving the PSR Cache code under the Joomla namespace for now, and then get v2 ready with the official Cache PSR once it's finalized.

**Comment by ghost** [comment]
I don't think there are any b/c headaches since the vendor directory isn't replicated to https://github.com/joomla/joomla-framework-cache - so composer installs of cache are currently broken to begin with[which was how I discovered the issue when I tried to use only a few specific framework components in a project].

Because it is defined as a repository override, in theory when psr cache actually is released, removing just the repository lines will allow it to pull in the real packagist package.

Definitely would be better to create a joomla/psr-cache repo as the placeholder instead - that just requires someone with authority to do so..  [and in the meantime, for my own personal use I can merge refs/pull/334/head from the framework into my repository via composer as my reference commit. :-)]

**Comment by ianmacl** [comment]
We're going to rework composer.json so we'll look at something like this later.  Closing.

**Comment by ghost** [comment]
Out of curiosity, where is the conversation about reworking composer taking place?  The mailling list seems pretty dead.

**Comment by dongilbert** [comment]
During the architecture sprint this weekend. Had most of the Framework team together.

**Pull request closed without merging**